packages/core/src/authenticatedFetch/fetchFactory.spec.ts

```diff
@@ -98,6 +98,17 @@ const mockFetch = (response: Response, url: string) => {
 };
 
 describe("buildAuthenticatedFetch", () => {
+  const spyTimeout = jest.spyOn(global, "setTimeout");
+
+  afterEach(() => {
+    // Clear the latest timeout to avoid dangling open handles.
+    // FIXME: Should just use fake timers, but that chokes on recursive calls.
+    const handle = spyTimeout.mock.results[spyTimeout.mock.results.length - 1];
+    if (handle !== undefined) {
+      (handle.value as ReturnType<typeof setTimeout>).unref();
+    }
+  });
```

Comment:
Can this be changed with the usage of mock timers? https://jestjs.io/docs/timer-mocks — err, totally blanked on the comment; Let's open a Jira ticket to fix this.

Reply:
I wonder if the issue you faces was due to https://jestjs.io/docs/timer-mocks#run-pending-timers

i.e., `jest.runOnlyPendingTimers();` vs `jest.runAllTimers();`

Reply:
Unfortunately no it didn't appear to be the issue. Basically it looks like Jest handles the timers being set by `setTimeout`, but not the ones set in the callback passed to `setTimeout`, resulting in the second round of timers not to be mocked.